internal/validation/validation.go

```diff
@@ -508,7 +508,7 @@ func validateLiteral(c *opContext, l common.Literal) {
 func validateValueType(c *opContext, v common.Literal, t common.Type) (bool, string) {
 	if v, ok := v.(*common.Variable); ok {
 		for _, op := range c.ops {
-			if v2 := op.Vars.Get(v.Name); v != nil {
+			if v2 := op.Vars.Get(v.Name); v != nil && v2 != nil {
```

Comment:
Why didn't you just replace `v != nil`? It's not needed. I'm guessing it was originally a typo and @neelance meant to write `v2` instead. `v` is guaranteed not to be `nil` by the time the `v != nil` check executes, because if it were, `v.Name` would have already panicked. So, this code is now equivalent to:

```Go
if v2 := op.Vars.Get(v.Name); true && v2 != nil {
```

If you do want to check that `v != nil`, consider moving it to line 509:

```Go
if v, ok := v.(*common.Variable); ok && v != nil {
```

/cc @dominikh Do you think this could be detected by a static check?